airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -104,105 +108,220 @@ private void saveAttemptSyncConfig(final long jobId, final int attemptNumber, fi
         "set attempt sync config");
   }
 
+  private IntegrationLauncherConfig getSourceIntegrationLauncherConfig(final long jobId,
+                                                                       final int attempt,
+                                                                       final ConfigType configType,
+                                                                       final JobSyncConfig config,
+                                                                       final StandardSourceDefinition sourceDefinition,
+                                                                       final JsonNode sourceConfiguration)
+      throws IOException {
+    final ConfigReplacer configReplacer = new ConfigReplacer(LOGGER);
+
+    final IntegrationLauncherConfig sourceLauncherConfig = new IntegrationLauncherConfig()
+        .withJobId(String.valueOf(jobId))
+        .withAttemptId((long) attempt)
+        .withDockerImage(config.getSourceDockerImage())
+        .withProtocolVersion(config.getSourceProtocolVersion())
+        .withIsCustomConnector(config.getIsSourceCustomConnector())
+        .withAllowedHosts(configReplacer.getAllowedHosts(sourceDefinition.getAllowedHosts(), sourceConfiguration));
+
+    if (!ConfigType.RESET_CONNECTION.equals(configType)) {
+      sourceLauncherConfig.setAllowedHosts(configReplacer.getAllowedHosts(sourceDefinition.getAllowedHosts(), sourceConfiguration));
```

Comment:
This looks redundant, I think we can remove this if.

Reply:
Good catch, it wasn't redundant but the mistake is actually the line above. We don't want to set allowed hosts on resets - see https://github.com/airbytehq/airbyte/pull/21935